sympy/physics/units/tests/test_quantities.py

```diff
@@ -221,3 +222,9 @@ def test_Quantity_derivative():
     assert diff(x**3*meter**2, x) == 3*x**2*meter**2
     assert diff(meter, meter) == 1
     assert diff(meter**2, meter) == 2*meter
+
+
+def test_sum_of_incompatible_quantities():
+    raises(ValueError, lambda: meter + second)
+    raises(ValueError, lambda: 2 * meter + second)
+    raises(ValueError, lambda: 2 * meter + 3 * second)
```

Comment:
@asmeurer this is the test case where `Add` needs to know that `Mul` contains `Quantity` objects. Without that knowledge, it won't raise an exception for the summing of incompatible unit dimensions (length and time).